lib/susedistribution.pm

```diff
@@ -191,14 +191,16 @@ sub x11_start_program {
     # after 'ret' press we should wait in this case nevertheless
     wait_still_screen(3) unless ($args{no_wait} || ($args{valid} && $args{target_match} && !check_var('DESKTOP', 'kde')));
     return unless $args{valid};
+    my $target = ref $args{target_match} eq 'ARRAY' ? @{$args{target_match}} : $args{target_match};
     for (1 .. 3) {
-        assert_screen([ref $args{target_match} eq 'ARRAY' ? @{$args{target_match}} : $args{target_match}, 'desktop-runner-border'],
-            $args{match_timeout}, no_wait => $args{match_no_wait});
+        assert_screen(["$target", 'desktop-runner-border'], $args{match_timeout}, no_wait => $args{match_no_wait});
         last unless match_has_tag 'desktop-runner-border';
         wait_screen_change {
             send_key 'ret';
         };
     }
+    # asserting program came up properly
+    assert_screen "$target", 0;
```

Comment:
Hm, let's think about if this makes sense: We already have in L196 an assert_screen which needs to match target or desktop-runner-border. And in case of !desktop-runner-border the previous for-loop exits. But we do not check if the for-loop exits because we are out of tries or because we actually matched the target. I think it would look better if we do not check explicitly a second time. How about we use a more same approach as in other cases? E.g.

```perl
for (1 .. 3) {
    assert_screen(["$target", 'desktop-runner-border'], $args{match_timeout}, no_wait => $args{match_no_wait});
    last unless match_has_tag 'desktop-runner-border';
    wait_screen_change { send_key 'ret' };
}
die "Did not find target needle match for tag '$target'" if match_has_tag 'desktop-runner-border';
```

Reply:
Good approach, will implement it

Reply:
The point here is to show it clearly in openQA and don't see candidates. If we check previous screen we will have desktop-runner-border which matches, so this is a drawback of dying. Also, please adjust text as target can be array ref and we don't want to report array ref address, similarly s/tag/tag(s)/

Reply:
@rwx788 so are you saying my approach is flowed or the one from @DrMullings ?

Reply:
It's obviously subjective, but I would expect that for the reviewer it would be easier to understand if there is a step with expected needles which didn't match instead of a message referring to previous step which actually matched, but for needle which is not expected there. So she or he will have to look into the code. This is the cons of 'die'. Cons of current solution is additional assert_screen. Maybe we should rethink this loop then.

Reply:
For me as a reviewer I would prefer to see the short text explaining why it failed over an assert_screen which I might need to investigate further to understand whats going on

Reply:
So why it's not in your PR already then? =) I've just expressed my opinion.